puppet/modules/slave/manifests/packaging/rpm.pp

```diff
@@ -114,4 +116,10 @@
   package { ['dnf', 'dnf-plugins-core']:
     ensure => present,
   }
+
+  secure_ssh::rsync::uploader_key { 'yumstage':
+    user       => $user,
+    dir        => "${workspace}/staging_key",
```

Comment:
One thing to note: for other things I've also taken the approach of manually adding this key to Jenkins as a real secret and then use the Jenkins built in functionality. Perhaps worth considering here too.